### PR TITLE
 feat: implemented Ifftshift and corresponding test function

### LINKED ISSUE #23801: ifftshift

### PR DESCRIPTION
**Pull request by VaishnaviMudaliar**
# PR Description

## Related Issue

Closes #23801 

## Checklist

- [X] Did you add a function?
- [X] Did you add the tests?
- [X] Did you run your tests and are your tests passing?
- [X] Did pre-commit not fail on any check?
- [X] Did you follow the steps we provided?

### CODE REVIEW

**Comment by VaishnaviMudaliar** [comment]
Hi @KareemMAX  , kindly review the PR and let me know if any changes are needed , thanks!

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py [hunk 1]
```diff
@@ -244,6 +244,17 @@ def test_jax_numpy_ifft2(
...
+@handle_frontend_test(
+    fn_tree="jax.numpy.fft.ifftshift",
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"), shape=(4,), array_api_dtypes=True
+    ),
+)
+def test_jax_numpy_ifftshift(
+    dtype_and_x, backend_fw, frontend, test_flags, fn_tree, on_device
+):
+    input_dtype, arr = dtype_and_x
```
Comment (Ishticode) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:257]: the main function test call is missing

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py [hunk 2]
```diff
@@ -254,6 +265,7 @@ def test_jax_numpy_ifftn(
 ):
     input_dtype, x, s, axes, norm = dtype_and_x
 
+ main
```
Comment (Ishticode) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:268]: this shouldn't be here

File: ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py [hunk 3]
```diff
@@ -262,12 +274,17 @@ def test_jax_numpy_ifftn(
         fn_tree=fn_tree,
         on_device=on_device,
         test_values=True,
+
+        x=arr[0],
```
Comment (Ishticode) [inline ivy_tests/test_ivy/test_frontends/test_jax/test_numpy/test_fft.py:278]: This seems to modify a different test than the one we are adressing in the PR

**Review by Ishticode — CHANGES_REQUESTED** [review]
Hi @VaishnaviMudaliar 
Thank you very much for the PR. 
Could you exclude your changes to docs/demos and please also take a look at few requested changes on the code below. Thank you very much :)

**Pull request closed without merging**